api/base/parsers.py

```diff
@@ -15,6 +17,29 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    def flatten_relationships(self, relationships):
+        """
+        Flattens relationships dictionary which has information needed to create related resource objects.
+
+        Validates that formatting of relationships dictionary is correct.
+        """
+        if not isinstance(relationships, dict):
+                raise ParseError()
```

Comment:
I try to be pretty thorough with test coverage, but I missed that one.  I tested for if `relationships` was missing, if `relationships` was empty, but not if `relationships` was a list with content inside.  I will add this.